errors.go

```diff
@@ -152,6 +152,34 @@ func ReplaceNativeMessage(e error, msg string) Error {
 	}
 }
 
+// AsType "casts" an input error as a target merry.Error such that the returned merry Error evaluates
+// to true when operating Is(newErr, targetErr). If e is nil, return nil. If targetType is nil,
+// A wrapped merry Error of e is returned. The key value pairs of the targetType is kept and then
+// appended with key value map from the input error. Stack trace of the original input error is maintained.
+func AsType(e error, targetType Error) (Error, bool) {
```

Comment:
I don't feel it's necessary to return a success bool as the 2nd return parameter. If the client calls this function with nil for either or both arguments, it should expect either a nil or the same merry wrapped error. Garbage in - garbage out. Adding a 2nd parameter forces all clients of this function to have to handle the case where the AsType "fails" and I don't really see the usefulness of this.

Reply:
Having the second parameter is akin to doing `w, ok := e.(*merryErr)` in which the client is notified that the casting worked or not. 

For the case that `targetType` is nil, it is unclear what the intended behavior should be. The client expects the returned `Error` to be of `targetType`, instead of asking the client to do a `Is()` comparison, we can simplify return `false` to indicate that the cast did not work as wished. This is the same idea with the case of input `e` being nil.

Reply:
Either of the design looks good to me. It all depends on what the client expect it to be. @alexherns seems to prefer just returning nil?

Reply:
@MayaLi -- to me the difference between this function and type casting is that in golang, handling the `ok` return argument is optional and by doing so breaks with almost every other convention in the language. I.e. both of these statements are legal:
```go
w, ok := e.(*merryErr)
w := e.(*merryErr)
```
For better or worse, we aren't allowed to write functions in that way, which means that, as @wangdrew suggests, every consumer would have to at least receive the 2nd return parameter. And sure, all consumers _could_ just do `err, _ = merry.AsType(err, OtherErrorType)`, but I think that having the 2nd return parameter there would give me the feeling that I should do something with it, which adds additional complexity to my code every time I call `merry.AsType`.

Reply:
Two return args also means you can't use chaining, which is a pattern generally supported by merry.

Reply:
Chaining is a good reason for a single parameter return. I just read merry code again and looks like the functions account for nil input. That's safe enough.